modules/targetanalysis/src/target_loader.cpp

```diff
@@ -66,61 +73,63 @@ using namespace std;
 * }
 */
 
-/*
-class targetLoader{
-      targetLoader(string f){
-             file = f;     
-      }
-      public:
-             TargetParameters getParameters();
-             class TargetParameters{
-                   public:
-                          string* parametersTypes;
-                          Parameter* parameters;
-                          unsigned int parameterCount;
-                          class Parameter{
-                                public:
-                                       bool enabled;
-                                       double value;
-                                       unsigned int confidence;
-                          }
-             }
-      private:
-              string file;
-              readFile(string fileLocation);
-              
-}
-
-
-
 
+/*
 TargetParameters targetLoader:getParameters(){
        return 0;
 }
-
-char* targetLoader:readFile(string fileLocation){
+*/
+bool TargetLoader::readFile(string fileLocation, char* data){
      ifstream jFile;
-     char* memblock;
      
      jFile.open(fileLocation, ios::binary | ios::ate);
-     if (file.is_open())
-     {
-        size = file.tellg();
-        memblock = new char[size];
-        file.seek(0, ios::beg);
-        file.read(memblock,size);
-        file.close();
+     if (jFile.is_open()) {
+        int size;
+	size = jFile.tellg();
+        data = new char[size];
+        jFile.seekg(0, ios::beg);
+        jFile.read(data,size);
+        jFile.close();
+     }
+     else{
+	cout << "Could not open file: " << fileLocation << "\n";
```

Comment:
Use BOOST_LOG_TRIVIAL(error) rather than cout (or cerr) to log errors or messages. That way we can filter messages.
It occurs to me that we don't actually have the fact that we're using this documented anywhere, so I can hardly blame you for not using Boost.Log.